src/api/dataops/filteringCondition.ts

```diff
@@ -49,47 +47,73 @@ export class FilteringCondition<U> implements ICondition {
   }
 
   public compile() {
-    return { field: this.field, operator: this.operator, values: this.values, type: this.Type };
+    return [
+      { field: this.field },
+      this.operator,
+      this.value,
+    ];
   }
 }
 
 /**
  * @internal
  */
 export class CompositeFilteringCondition implements ICondition {
-  private logicalOperatorType: string;
+  private logicalOperatorType: LogicalOperator;
   private conditions: ICondition[];
 
-  constructor(filteringCondition: ICondition, logicalOperatorType: string) {
-    this.conditions = [];
-    this.conditions.push(filteringCondition);
+  constructor(filteringCondition: ICondition, logicalOperatorType: LogicalOperator) {
+    this.conditions = [filteringCondition];
     this.logicalOperatorType = logicalOperatorType;
   }
 
-  public get Type(): string{
+  public get type(): LogicalOperator {
     return this.logicalOperatorType;
   }
 
-  public set Type(type: string) {
+  public set type(type: LogicalOperator) {
     this.logicalOperatorType = type;
   }
 
   public or(condition: ICondition): CompositeFilteringCondition {
-    condition.Type = "or";
-    this.conditions.push(condition);
-    return this;
+    return this.appendCondition(condition, "or");
   }
 
   public and(condition: ICondition): CompositeFilteringCondition {
-    condition.Type = "and";
-    this.conditions.push(condition);
-    return this;
+    return this.appendCondition(condition, "and");
   }
 
   public compile() {
-    return {
-      conditions: this.conditions.map( (condition: ICondition) => condition.compile()),
-      type: this.logicalOperatorType,
-     };
+    return this.conditions.reduce(this.toCompiledConditions, []);
+  }
+
+  private toCompiledConditions(expressions: any[], condition: ICondition) {
+    const appendNestedConditions = () => {
+      const compiledCondition = condition.compile();
+      const SINGLE_CONDITION_LENGTH = 3;
+
+      if (compiledCondition.length === SINGLE_CONDITION_LENGTH) {
```

Comment:
as I understood `compile()` method, there is always 3 elements in compiled condition. In what cases it can be different?

Reply:
example: 

`field("name").isEqualTo("andrey")` should "compile" to `[{ field: "name" }, "=", "andrey"]` this is a compiled "single condition", an array of 3 elements. 

when you do:
`field("name").isEqualTo("andrey").and(field("age").isEqualTo(18))` should "compile" to `[{ field: "name" }, "=", "andrey", "and", { field: "age" }, "=", 18]` that is an array with more than 3 elements. 

That's why I spread the "single condition". That's a bit tricky, if you have another approach feel free to share your thoughts

Reply:
What about this one?
```
export interface ICondition {
  simple: boolean;
  ...
}

private toCompiledConditions(expressions: any[], condition: ICondition) {
    if (expressions.length) {
      // append connector
      expressions.push(condition.type);
    }

    const compiledCondition = condition.compile();

    if (condition.simple) {
      expressions.push(...compiledCondition);
    } else {
      expressions.push(compiledCondition);
    }

    return expressions;
  }
```

Reply:
I like it, that's fixed